gcloud/rest/auth/token.py

```diff
@@ -19,77 +29,208 @@
-        if 'error' in content:
-            raise Exception('{}'.format(content))
+        now = datetime.datetime.utcnow()
+        delta = (now - self.access_token_acquired_at).total_seconds()
+        if delta <= self.access_token_duration / 2:
+            return
 
-        self.age = datetime.datetime.now()
-        self.expiry = int(content['expires_in'])
-        self.value = content['access_token']
+        self.acquire_access_token()
 
     def ensure(self):
-        if not self.value:
-            log.debug('acquiring initial token')
-            self.acquire()
-            return
+        # type: () -> None
+        warnings.warn('Token.ensure() is deprecated',
+                      'please use Token.ensure_token()',
+                      DeprecationWarning)
+        self.ensure_token()
+
+    def _refresh_authorized_user(self, timeout):
+        # type: (int) -> requests.Response
+        payload = urlencode({
+            'grant_type': 'refresh_token',
+            'client_id': self.service_data['client_id'],
+            'client_secret': self.service_data['client_secret'],
+            'refresh_token': self.service_data['refresh_token'],
+        }, quote_via=quote_plus)
+        with self.google_api_lock:
+            return self.session.post(self.token_uri, data=payload,
+                                     headers=REFRESH_HEADERS, timeout=timeout)
+
+    def _refresh_gce_metadata(self, timeout):
+        # type: (int) -> requests.Response
+        with self.google_api_lock:
+            return self.session.get(self.token_uri,
+                                    headers=GCE_METADATA_HEADERS,
+                                    timeout=timeout)
 
-        now = datetime.datetime.now()
-        delta = (now - self.age).total_seconds()
+    def _refresh_service_account(self, timeout):
+        # type: (int) -> requests.Response
+        now = int(time.time())
+        assertion_payload = {
+            'aud': self.token_uri,
+            'exp': now + GCLOUD_TOKEN_DURATION,
+            'iat': now,
+            'iss': self.service_data['client_email'],
+            'scope': self.scopes,
+        }
 
-        if delta > self.expiry / 2:
-            log.debug('requiring token with expiry %d of %d / 2', delta,
-                      self.expiry)
-            self.acquire()
+        # N.B. algorithm='RS256' requires an extra 240MB in dependencies...
+        assertion = jwt.encode(assertion_payload,
+                               self.service_data['private_key'],
+                               algorithm='RS256')
+        payload = urlencode({
+            'assertion': assertion,
+            'grant_type': 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+        })
+        with self.google_api_lock:
+            return self.session.post(self.token_uri, data=payload,
+                                     headers=REFRESH_HEADERS, timeout=timeout)
+
+    @backoff.on_exception(backoff.expo, Exception, max_tries=1)  # type: ignore
```

Comment:
Why change this to `1` instead of `5` tries? Not much point of having a backoff handler unless it has at least one retry

Reply:
Ahh, this is my bad. When running the integration tests, I was getting a '429 Client Error: Too Many Requests for url' error, which I pinged you about. So this was just me trying to see if the max_tries was causing this. Will revert this back to 5